cogs/lookingforgroup.py

```diff
@@ -0,0 +1,899 @@
+        self.bot.add_view(PlayersSubmissionView(bot=self.bot))
+        self.bot.add_view(CommunitySubmissionView(bot=self.bot))
+        self.bot.add_view(PaidSubmissionView(bot=self.bot))
+
+        self.bot.add_view(PostedView(bot=self.bot))
+        self.bot.add_view(ModResetView(bot=self.bot))
+
+        self.cleanup_timers.start()
+
+        # Try to get the autodelete loop timing
+        auto_delete_cog = self.bot.get_cog("AutoDelete")
+        if auto_delete_cog:
+            loop: disnake.ext.tasks.Loop = auto_delete_cog.deleter
+            self.loop_offset = loop.next_iteration
+
+        print(f"Looking for group offset set as {self.loop_offset}...")
+
+    def cog_unload(self):
+        self.cleanup_timers.cancel()
+
+    @tasks.loop(hours=168)
+    async def cleanup_timers(self):
+        await self.bot.wait_until_ready()
+
+        print(f"Cleaning up old /looking-for timers...")
+
+        def expired_timer(pair) -> bool:
+            key, value = pair
+            if value[0] <= now:
+                return False
+            return True
+
+        new_lfg_timers = {}
+        now = datetime.now().timestamp()
+
+        for user_id, timers in self.lfg_timers.items():
+            timers = dict(filter(expired_timer, timers.items()))
+            if timers:
+                new_lfg_timers[user_id] = timers
+
+        self.lfg_timers = new_lfg_timers
+        self.bot.db.jset("lookingforgroup", self.lfg_timers)
+
+    def base_embed(self, title: str, inter: disnake.ApplicationCommandInteraction):
+        embed = disnake.Embed(timestamp=self.loop_offset + timedelta(days=7))
+        embed.title = title
+        embed.description = inter.author.mention
+        embed.set_footer(text=f"User ID: {inter.author.id} - Expires on")
+
+        return embed
+
+    async def post_cooldown(self, inter: disnake.ApplicationCommandInteraction, timer_type: str):
+        """Check the cooldown on a posting type for a specific user."""
+
+        timers: dict = self.lfg_timers.get(str(inter.author.id), {})
+        timer, message_id = timers.get(timer_type, (0, 0))
+
+        if not timer:
+            return False
+        elif timer <= datetime.now().timestamp():
+            timers.pop(timer_type)
+            self.lfg_timers[inter.author.id] = timers
+            self.bot.db.jset("lookingforgroup", self.lfg_timers)
+            return False
+
+        channel = await inter.guild.fetch_channel(CHANNELS.get(timer_type))
+
+        try:
+            message = await channel.fetch_message(message_id)
+            message_link = f"\n\nYou can see your previous posting [Here]({message.jump_url})."
+        except disnake.errors.NotFound:
+            message_link = ""
+
+        await inter.send(
+            f"You have to wait until <t:{timer}> (<t:{timer}:R>) to post again.{message_link}",
+            ephemeral=True,
+            suppress_embeds=True,
+        )
+
+        return True
+
+    @commands.user_command(name="Reset - DM")
```

Comment:
Should these have a permission requirement by default?

Reply:
They sure probably should!